### PR TITLE
MLHR-1687: removed threshold property from block reader

### PR DESCRIPTION
**Pull request by chandnisingh**

### CODE REVIEW

**Comment by PramodSSImmaneni** [comment]
Chandni what is the replacement for this?

**Comment by chandnisingh** [comment]
Upstream operator is responsible for slowing down if required by the application.

File: library/src/main/java/com/datatorrent/lib/io/block/AbstractBlockReader.java [hunk 1]
```diff
@@ -122,10 +115,7 @@
     @Override
     public void process(B block)
     {
-      blockQueue.add(block);
-      if (blocksPerWindow < threshold) {
-        processHeadBlock();
-      }
+      processBlockMetadata(block);
```
Comment (243826) [inline library/src/main/java/com/datatorrent/lib/io/block/AbstractBlockReader.java]: It probably makes sense to remove this method redirection as well.

Comment (chandnisingh) [inline library/src/main/java/com/datatorrent/lib/io/block/AbstractBlockReader.java]: no it is there so the concrete implementations can override it if needed.  For eg. if the processing of block fails then I would like to capture that information and they way to do it is by overriding this method.

**Comment by PramodSSImmaneni** [comment]
>>> Upstream operator is responsible for slowing down if required by the application.

Sounds good. If it not possible to throttle the input may be we can break within blocks, so end window boundaries within blocks (at record level). I am doing elsewhere in a different operator in another application.

**Pull request merged**